core/src/link_resolver.rs

```diff
@@ -90,25 +87,26 @@ async fn select_fastest_client_with_stat<'a>(
     }
 
     Err(err.unwrap_or_else(|| {
-        format_err!("No IPFS clients were supplied to handle the call to object_stat")
+        format_err!(
+            "No IPFS clients were supplied to handle the call to object_stat. File: {}",
+            path
+        )
     }))
 }
 
-// Returns an error if the stat is bigger than the `max_file_bytes`
-fn restrict_file_size(
-    path: &str,
-    stat: &ObjectStatResponse,
-    max_file_bytes: &Option<u64>,
-) -> Result<(), failure::Error> {
-    if let Some(max_file_bytes) = max_file_bytes {
-        if stat.cumulative_size > *max_file_bytes {
-            return Err(format_err!(
-                "IPFS file {} is too large. It can be at most {} bytes but is {} bytes",
-                path,
-                max_file_bytes,
-                stat.cumulative_size
-            ));
-        }
+/// Returns an error if the stat is bigger than the `MAX_IPFS_MAP_FILE_SIZE_VAR`
+fn restrict_file_size(path: &str, stat: &ObjectStatResponse) -> Result<(), failure::Error> {
+    // FIXME: Having an env variable here seems like a problem for consensus.
+    // Index Nodes should not disagree on whether the file should be read.
+    let max_file_size =
+        read_u64_from_env(MAX_IPFS_MAP_FILE_SIZE_VAR).unwrap_or(DEFAULT_MAX_IPFS_MAP_FILE_SIZE);
```

Comment:
Ah, this unfortunately doesn't work. The file size limits for `ipfs.cat` and `ipfs.map` are deliberately split up into two environment variables, as `ipfs.map` is intended for larger files than `ipfs.cat`.

Plus both env vars are already in use, so we can't simply drop the one for `ipfs.cat`.

Reply:
So I think we still have to pass the file size in.